src/figuresClasses.ts

```diff
@@ -1,19 +1,93 @@
-export interface Figure {
-
+function isNotZeroSides(...sides: number[]): boolean {
+  return sides.every((side) => side >= 0);
 }
 
-export class Triangle {
+type Color = 'red' | 'blue' | 'green';
+
+export enum FigureShapes {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+export interface Figure {
+  color: Color;
+  shape: FigureShapes;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  public shape = FigureShapes.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (!isNotZeroSides(a, b, c)) {
+      throw new Error('Sorry , all sides must be greater than zero !!!');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error(
+        'Sorry , 1 side can not be greater then sum of 2 other(((',
+      );
+    }
+  }
+
+  getArea(): number {
+    const semiPerimeter: number = (this.a + this.b + this.c) / 2;
 
+    const area = Math
+      .sqrt(semiPerimeter
+        * (semiPerimeter - this.a)
+        * (semiPerimeter - this.b)
+        * (semiPerimeter - this.c));
+
+    return Math.floor(area * 100) / 100;
```

Comment:
you round down to hundredths on each getArea, maybe better create a separate method for it?